SAMLPortal/Controllers/AppController.cs

```diff
@@ -0,0 +1,219 @@
+			{
+				//Log the error (uncomment ex variable name and write a log.
+				ModelState.AddModelError("", "Unable to save changes. " +
+					"Try again, and if the problem persists " +
+					"see your system administrator.");
+			}
+			return View(appli);
+		}
+
+		// GET: App/Edit/5
+		[AllowAnonymous]
+		[Route("Edit/{id}")]
+		public async Task<IActionResult> Edit(int? id)
+		{
+			if (id == null)
+			{
+				return NotFound();
+			}
+
+			SAMLPortalContext context = new SAMLPortalContext();
+			var app = await context.App.FindAsync(id);
+			if (app == null)
+			{
+				return NotFound();
+			}
+			return View("Edit", app);
+		}
+
+		// POST: App/Edit/5
+		[HttpPost]
+		[ValidateAntiForgeryToken]
+		[Route("Edit/{id}")]
+		public async Task<IActionResult> EditPost(int? id)
+		{
+			if (id == null)
+			{
+				return NotFound();
+			}
+			SAMLPortalContext context = new SAMLPortalContext();
+			var appToUpdate = await context.App.FirstOrDefaultAsync(s => s.Id == id);
+			if (await TryUpdateModelAsync<App>(
+				appToUpdate,
+				"",
+				s => s.Name,
+				s => s.Description,
+				s => s.Role,
+				s => s.Enabled,
+				s => s.MetadataURL,
+				s => s.Issuer,
+				s => s.SingleSignOnDestination,
+				s => s.SingleLogoutResponseDestination))
+			{
+				try
+				{
+					await context.SaveChangesAsync();
+					return Redirect("/");
+				}
+				catch (DbUpdateException /* ex */)
+				{
+					//Log the error (uncomment ex variable name and write a log.)
+					ModelState.AddModelError("", "Unable to save changes. " +
+						"Try again, and if the problem persists, " +
+						"see your system administrator.");
+				}
+			}
+			return View("Edit", appToUpdate);
+		}
+
+		// GET: App/Delete/5
+		[Route("Delete/{id}")]
+		public async Task<IActionResult> Delete(int? id, bool? saveChangesError = false)
+		{
+			if (id == null)
+			{
+				return NotFound();
+			}
+			SAMLPortalContext context = new SAMLPortalContext();
+			var app = await context.App.AsNoTracking().FirstOrDefaultAsync(m => m.Id == id);
+			if (app == null)
+			{
+				return NotFound();
+			}
+
+			if (saveChangesError.GetValueOrDefault())
+			{
+				ViewData["ErrorMessage"] =
+					"Delete failed. Try again, and if the problem persists " +
+					"see your system administrator.";
+			}
+			return View(app);
+		}
+
+		// POST: App/Delete/5
+		[HttpPost]
+		[ValidateAntiForgeryToken]
+		[Route("Delete/{id}")]
+		public async Task<IActionResult> DeleteConfirmed(int id)
+		{
+			SAMLPortalContext context = new SAMLPortalContext();
+			var app = await context.App.FindAsync(id);
+			if (app == null)
+			{
+				return Redirect("/");
+			}
+
+			try
+			{
+				context.App.Remove(app);
+				await context.SaveChangesAsync();
+				return RedirectToAction(nameof(Index));
+			}
+			catch (DbUpdateException /* ex */)
+			{
+				//Log the error (uncomment ex variable name and write a log.)
+				return RedirectToAction(nameof(Delete), new { id = id, saveChangesError = true });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the redundant 'id ='.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5434886)